### PR TITLE
feat: add swift code to lcci problems: No.01.02,02.05

### PR DESCRIPTION
**Pull request by klever34**
- I noticed there weren't any implementations in Swift, so I decided to add some.
- If this is approved, I'll go ahead with writing for the other implementations.

### CODE REVIEW

**Comment by idoocs** [comment]
🤭 感谢你的提交，请检查你的改动是否符合以下项目规范。

### 1. 格式化

我们项目中各种编程语言代码（包括文档）所采用的格式化工具不同，提交 pr 之前必须确保代码、文档正确格式化。

-   .{md,js,ts,php,sql,rs} 采用 prettier
-   .{c,cpp,java} 采用 clang-format
-   .{py} 采用 black
-   .{go} 采用 gofmt
-   其它待完善

### 2. Git 提交信息

我们项目遵循 [AngularJS Git Commit Message Conventions](https://docs.google.com/document/d/1QrDFcIiPjSLDn3EL15IJygNPiHORgU1_OOAqWjiDU5Y/edit#) 规范，我们希望你的提交信息尽可能与项目保持一致。

-   新增或修改题解：feat: add/update solution(s) to lc problem(s): No.xxxx
-   修复错误：fix: xxxx
-   日常维护：chore: xxx

### 3. 其它补充

新增题解及代码时，需要创建 Solution.xxx 源代码文件（如果已存在，请确认算法是否更优，是则覆盖已有算法代码），同时，需要在 README.md 以及 README_EN.md 中添加对应的代码片段（英文文件中不要出现中文注释）
另外，编码风格（比如变量、函数的命名），尽量跟项目已有代码保持一致。 

---

🤭 Thank you for your contribution. Please check if your changes comply with the following project specifications. 

### 1. Formatting

We use different formatting tools for various programming languages (including documentation) in our project. You must ensure that the code and documentation are correctly formatted before submitting a pr.

-   .{md,js,ts,php,sql,rs} use prettier
-   .{c,cpp,java} use clang-format
-   .{py} use black
-   .{go} use gofmt
-   Others to be improved

### 2. Git Commit Message

Our project follows the [AngularJS Git Commit Message Conventions](https://docs.google.com/document/d/1QrDFcIiPjSLDn3EL15IJygNPiHORgU1_OOAqWjiDU5Y/edit#). We hope that your submission information is as consistent as possible with the project.

-   Add or modify solutions: feat: add/update solution(s) to lc problem(s): No.xxxx
-   Fix errors: fix: xxxx
-   Routine maintenance: chore: xxx

### 3. Other notes

When adding solutions and code, you need to create a Solution.xxx source code file (if it already exists, please confirm whether the algorithm is better, if yes, overwrite the existing algorithm code), and at the same time, you need to add the corresponding code snippets in README.md and README_EN.md (do not have Chinese comments in the English file)
In addition, the coding style (such as the naming of variables and functions) should be as consistent as possible with the existing code in the project.

**Comment by yanglbme** [comment]
Hi @klever34 thanks for your contribution :)

For the first question, the Swift code template is as follows:

```swift
class Solution {
    func isUnique(_ astr: String) -> Bool {

    }
}
```

And for the second question, the Swift code template is:

```swift
class Solution {
    func CheckPermutation(_ s1: String, _ s2: String) -> Bool {

    }
}
```

You need to complete the code based on the code provided here, and then paste the complete code into the corresponding Solution.swift file for each question. Additionally, there are two points to note:

1. You need to add the corresponding Swift code blocks in README.md and README_EN.md.
2. Your code implementation should preferably be consistent with existing code implementations in other languages, such as variable names, function names, etc."

Let me know if you need any further assistance!

**Comment by klever34** [comment]
Hi @yanglbme 
Kindly take a look at the PR again.
Do let me if I need to adjust anything.

**Comment by yanglbme** [comment]
@klever34 

According to our project's guidelines, the code in Solution.swift should be identical to the Swift code snippet in the README. Therefore, please update the code for 02.05 as follows:

```swift
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init(_ val: Int) {
 *         self.val = val
 *         self.next = nil
 *     }
 * }
 */
class Solution {
    func addTwoNumbers(_ l1: ListNode?, _ l2: ListNode?) -> ListNode? {
        var carry = 0
        let dummy = ListNode(0)
        var current: ListNode? = dummy
        var l1 = l1, l2 = l2

        while l1 != nil || l2 != nil || carry != 0 {
            let sum = (l1?.val ?? 0) + (l2?.val ?? 0) + carry
            carry = sum / 10
            current?.next = ListNode(sum % 10)
            current = current?.next
            l1 = l1?.next
            l2 = l2?.next
        }

        return dummy.next
    }
}
```

Additionally, you need to place the Swift code for problem 01.02 into Solution 1 of README.md and README_EN.md, rather than Solution 2. It's preferable to use the same variable name `cnt` as in the existing solution, instead of `count`.

File: lcci/01.02.Check Permutation/README_EN.md [hunk 1]
```diff
@@ -242,6 +242,33 @@ impl Solution {
...
+            let index = Int(char.asciiValue! - Character("a").asciiValue!)
+            count[index] -= 1
+            if count[index] < 0 {
+                return false
+            }
+        }
+
+        return true
+    }
+}
```
Comment (yanglbme) [inline lcci/01.02.Check Permutation/README_EN.md:269]:
```suggestion
class Solution {
    func CheckPermutation(_ s1: String, _ s2: String) -> Bool {
        if s1.count != s2.count {
            return false
        }

        var cnt = Array(repeating: 0, count: 26)

        for char in s1 {
            let index = Int(char.asciiValue! - Character("a").asciiValue!)
            cnt[index] += 1
        }

        for char in s2 {
            let index = Int(char.asciiValue! - Character("a").asciiValue!)
            cnt[index] -= 1
            if cnt[index] < 0 {
                return false
            }
        }

        return true
    }
}
```

**Review by yanglbme — CHANGES_REQUESTED** [review]

**Review by yanglbme — APPROVED** [review]
Looks good, thanks 👍

**Pull request merged**